packages/block-library/src/image/image.js

```diff
@@ -412,19 +413,28 @@ export default function Image( {
 				return {};
 			}
 
-			const { getBlockBindingsSource } = unlock(
-				select( blockEditorStore )
-			);
+			const { getBlockBindingsSource, getBlockParentsByBlockName } =
+				unlock( select( blockEditorStore ) );
 			const {
 				url: urlBinding,
+				href: hrefBinding,
 				alt: altBinding,
 				title: titleBinding,
 			} = metadata?.bindings || {};
+			const hasParentPattern =
+				getBlockParentsByBlockName( clientId, 'core/block' ).length > 0;
 			return {
 				lockUrlControls:
 					!! urlBinding &&
 					getBlockBindingsSource( urlBinding?.source )
 						?.lockAttributesEditing === true,
+				lockHrefControls:
+					( !! hrefBinding &&
+						getBlockBindingsSource( hrefBinding?.source )
+							?.lockAttributesEditing === true ) ||
+					// Disable editing the link of the URL if the image is inside a pattern instance.
+					// This is a temporary solution until we support overriding the link on the frontend.
+					hasParentPattern,
```

Comment:
We don't support the `href` attribute yet, so `hrefBinding` should never exist. For sources like post meta, we disabled the href controls when the URL is bound because it didn't make sense to link to the image file or the attachment page. Should we just use something like this instead:

```js
lockHrefControls: hasParentPattern
```

And keep the URL conditional [later](https://github.com/WordPress/gutenberg/pull/58660/files#diff-a14b61c874095649b236b78413e805b4541277b24eb71ce350a17d9f2b6396b1R454-R456).

Reply:
Yes, that's what I meant 🙂